### ISSUE TITLE
utgovl step extremely slow

### ISSUE THREAD

**Issue by ersgupta** [marbl/canu#947]
I am running the unitigging step with heterozygous options (in continuation of #848 ).  The utgovl step seems to take like forever. There are 287 overlap jobs required. For each overlap job I am using 48 cores per process with 8 jobs in parallel (total cores being used 8*48). At an average I see only 2-4 jobs being completed per day.

I have the following queries:
1. Is it really expected to be this slow?
2. Is the computation resource not enough to speed this up?
3. Any way to increase the speed without compromising on the assembly quality?

Canu command:
```
~/apps/canu-1.7/Linux-amd64/bin/canu -assemble
 -d canu-pacbio-reutg
 -pacbio-corrected canu.trimmedReads.fasta.gz
 -p canu.utg
 gridOptions=' -N pb -l walltime=128:00:00'
 gridEngineMemoryOption='-l mem=MEMORY'
 gridEngineThreadsOption='-l nodes=1:ppn=THREADS'
 useGrid=remote
 genomeSize=1.4g
 correctedErrorRate=0.105
 "batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50"
```

**Comment by brianwalenz** [comment]
What did canu report for the hardware configuration?  For that genome size it shouldn't have picked more than 16 threads.  So I'm confused as to how it is configured.  Obvious questions: are the processes getting CPU time?  Is the machine paging?

With nanopore, we've seen one looooong read dominating the time for this stage, and that one read is processed by a single thread.  We haven't seen that with pacbio before.  Parameters `overlapper=mhap utgReAlign=true` are suggested for this case (http://canu.readthedocs.io/en/latest/faq.html).

**Comment by ersgupta** [comment]
You are right, it picked 16 threads per job. But that was much slower, hence I increased the number of threads to 48 (as Sergey also pointed out that more cores generally helps speed up). Does that not scale 3 times, or atleast >2 times?

About the ```overlapper=mhap utgReAlign=true``` option, I read in one of the posts that it compromises on contiguity. So, I did not use that option.

**Comment by ersgupta** [comment]
Another reason why I increased the cores was, at 16 cores I was using 240 hours (which I think was sufficient) of walltime and many jobs were going beyond this time and getting killed. So, rather than restarting a job after 10 days, I increased the cores to 48 and reduced walltime to 128 hours and till now no job is going beyond this. Some of them have reached ~110 hours.

For the file sizes of *ovb, till now 77/287 are complete and 20 files are > 1GB size.

**Comment by brianwalenz** [comment]
Ah, ok.  The repeat kmer thresholds are probably bonkers.  This is reported in `0-mercounts/*estMerThresh.out`:  (this from ecoli)
```
RAW MER COUNTS:
  distinct:      9666919 (different kmer sequences)
  unique:        4618002 (single-copy kmers)
  total:       169448169 (kmers in sequences)

Set maxCount to 114 (78 kmers), which will cover 99.75% of distinct mers and 98.25% of all mers.
Average number of kmers between count 89 and 140 is 23642
Found break in kmer coverage at 732
Set maxCount to 731 (1 kmers), which will cover 100.00% of distinct mers and 99.96% of all mers.
```
What's your coverage, and are you expecting this to be repeat rich?  I wouldn't mind seeing the `*.histogram` file.

Solution for you is to force a threshold with ovlMerThresh=<integer>, maybe something like 10*coverage (assuming your current threshold is ginormous).

**Comment by ersgupta** [comment]
Here it is:
```
cat unitigging/0-mercounts/canu.utg.ms22.estMerThresh.out
41352
...
[UNITIGGING/MERS]
--
--  22-mers                                                                                           Fraction
--    Occurrences   NumMers                                                                         Unique Total
--       1-     1 2396199691 *******************************************************************--> 0.6613 0.0516
--       2-     2 334047014 ********************************************************************** 0.7535 0.0660
--       3-     4 196656492 *****************************************                              0.7890 0.0743
--       5-     7  98268772 ********************                                                   0.8196 0.0848
--       8-    11  76685236 ****************                                                       0.8407 0.0961
--      12-    16  83447564 *****************                                                      0.8609 0.1125
--      17-    22  83713444 *****************                                                      0.8833 0.1387
--      23-    29  83412797 *****************                                                      0.9056 0.1741
--      30-    37  93612830 *******************                                                    0.9285 0.2224
--      38-    46  76586316 ****************                                                       0.9540 0.2908
--      47-    56  35744547 *******                                                                0.9737 0.3563
--      57-    67  14741410 ***                                                                    0.9826 0.3920
--      68-    79   9640201 **                                                                     0.9864 0.4105
--      80-    92   7080907 *                                                                      0.9890 0.4254
--      93-   106   5147084 *                                                                      0.9908 0.4382
--     107-   121   3967310                                                                        0.9922 0.4490
--     122-   137   3155611                                                                        0.9933 0.4586
--     138-   154   2517856                                                                        0.9942 0.4673
--     155-   172   2037266                                                                        0.9948 0.4751
--     173-   191   1684020                                                                        0.9954 0.4822
--     192-   211   1407131                                                                        0.9959 0.4887
--     212-   232   1197287                                                                        0.9962 0.4948
--     233-   254   1037071                                                                        0.9966 0.5005
--     255-   277    896258                                                                        0.9969 0.5059
--     278-   301    778472                                                                        0.9971 0.5110
--     302-   326    677699                                                                        0.9973 0.5158
--     327-   352    593633                                                                        0.9975 0.5204
--     353-   379    525980                                                                        0.9977 0.5247
--     380-   407    468586                                                                        0.9978 0.5288
--     408-   436    420150                                                                        0.9979 0.5328
--     437-   466    377580                                                                        0.9980 0.5366
--     467-   497    342518                                                                        0.9982 0.5403
--     498-   529    308695                                                                        0.9982 0.5438
--     530-   562    282724                                                                        0.9983 0.5472
--     563-   596    258258                                                                        0.9984 0.5505
--     597-   631    236518                                                                        0.9985 0.5537
--     632-   667    219600                                                                        0.9985 0.5569
--     668-   704    205340                                                                        0.9986 0.5599
--     705-   742    189451                                                                        0.9987 0.5630
--     743-   781    177559                                                                        0.9987 0.5659
--     782-   821    170610                                                                        0.9988 0.5688
--
--     3574269 (max occurrences)
-- 44041244847 (total mers, non-unique)
--  1227234316 (distinct mers, non-unique)
--  2396199691 (unique mers)
```
I copy the histogram from .report file. Please let me know if you need the complete histogram file.
We do not have estimate about the repeat content but as it is a huge genome (~1.4gb), I would expect highly repetitive regions. Also we have one assembly from falcon unzip, here is the busco stats from that: ```C:87.3%[S:57.6%,D:29.7%],F:2.9%,M:9.8%,n:1440```. 

Raw data coverage is ~77x, but here I am using only unitigging step with het parameters which has 33.21x coverage.

**Comment by brianwalenz** [comment]
Based on that, I'm fully expecting it picked a ginormous threshold.

To change ovlMerThresh, you need to remove the frequentMers file(s) from 0-mercounts (also removing the estMerThresh files wouldn't hurt).  If there aren't two large mcdat/mcidx files, kmers will need to be counted again (set saveMerCounts to keep these files).

**Comment by ersgupta** [comment]
Oh ok, so I kill the current jobs and delete the respective files from 0-mercounts. mcdat/mcidx are not there, so maybe delete the whole 0-mercounts? it does not take long either way, meryl took 2 hours last time.

And run with all the previous options plus ```ovlMerThresh=332```, right?

**Comment by skoren** [comment]
Given that level of duplication in FALCON-unzip I think your heterozygosity is quite high, over 1%. You can see this from the histogram, here are a lot of k-mers in the half-coverage (15x) count. In that case you should expect Canu to produce almost double the genome size (I'm not sure if 1.4g is the size of falcon primary or an independent estimate, if it's a falcon-unzip assembly it's probably an over-estimate). 

Given this, I don't think 30x corrected is going to be enough since you have only 15x per haplotype so I wouldn't waste time finishing this assembly. The heterozygous parameters on the FAQ adjust both the correction parameter and the unitigging parameters. The ```correctedErrorRate=0.105``` is actually not part of the heterozygous parameters since your unitigging parameters are set to split haplotypes but you should keep it if you have Sequel data. I would suggest to restart with the parameters:
```genomeSize=1.4g
 correctedErrorRate=0.105
 "batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50" 
corMhapSensitivity=normal corOutCoverage=100 ovlMerThreshold=500```

**Comment by ersgupta** [comment]
@skoren it is a highly heterozygous genome ~1.5% (based on genomescope from the default canu run). 1.4gb genome size is based on independent estimate based on ```2C=2.8 pg``` reported in a publication.

I am already running canu from scratch with het parameters as you had suggested ```corMhapSensitivity=normal corOutCoverage=200 correctedErrorRate=0.105 "batOptions=-dg 3 -db 3 -dr 1 -ca 500 -cp 50" genomeSize=1.4g OvlMerThreshold=300```. Shall I also add ```utgRealign=true``` to the parameters you mentioned?

**Comment by skoren** [comment]
Nope, you don't need ```utgReAlign``` if you don't also specify ```overlapper=mhap``` and I expect the ```ovlMerThreshold``` will make the default fast enough so you don't need either ```overlapper=mhap``` or ```utgReAlign=true```.

**Comment by ersgupta** [comment]
Cool, then I will let the current run from scratch to go on as it is (at correction right now). Shall I kill the re-unitigging with het parameters, as it does not add much?

About the parameters:
1. ```ovlMerThreshold =300``` or ```500```?
2. ```corOutCoverage=100``` or ```200```? This should not matter as the raw coverage is ~77x, right?

**Comment by skoren** [comment]
Yeah, I would trash it.

1. 500 because you should end up with more than 30x after correction with the other parameters
2. Either, doesn't matter since the raw coverage is lower than both.

**Comment by ersgupta** [comment]
Ohk.
1. So I kill the current from scratch run also? as I had used ```OvlMerThreshold=300```.

**Comment by skoren** [comment]
If it hasn't gotten past correction you can keep all the outputs and restart with with a higher ovlMerThreshold, it will just resume and it doesn't affect correction. I'm not sure how much 300 vs 500 matters either so you could just let it go.

**Comment by ersgupta** [comment]
correctReads.sh is almost over. So, maybe decide based on coverage I get after this step? and change for subsequent steps?

**Comment by skoren** [comment]
Yeah that make sense.

**Comment by ersgupta** [comment]
cool. thanks a lot. will keep you posted. :)

**Comment by ersgupta** [comment]
I get a coverage of 57.47 with ```ovlMerThreshold=300```, so need of repeating with 500 right? And I can keep it same for subsequent steps? or better change to 500?

**Comment by ersgupta** [comment]
@skoren in the other post #908 you mentioned the parameter ```OvlMerThreshold``` and here in the comments you mention ```ovlMerThreshold```, I want to clarify these both are same/similar, right?
I have used OvlMerThreshold in the current run.

Here are the outputs of *estMerThresh.out:
```
==> correction/0-mercounts/canu.het.ms16.estMerThresh.err <==
RAW MER COUNTS:
  distinct:   2146287554 (different kmer sequences)
  unique:        5545915 (single-copy kmers)
  total:    123120501546 (kmers in sequences)

Set maxCount to 1277 (6176 kmers), which will cover 99.75% of distinct mers and 83.84% of all mers.
Average number of kmers between count 1252 and 1303 is 7843929
Found break in kmer coverage at 96023
Set maxCount to 4400 (338 kmers), which will cover 99.96% of distinct mers and 91.43% of all mers.

==> trimming/0-mercounts/canu.het.ms22.estMerThresh.err <==
RAW MER COUNTS:
  distinct:   1933576568 (different kmer sequences)
  unique:      171109417 (single-copy kmers)
  total:     75753500874 (kmers in sequences)

Set maxCount to 1218 (5117 kmers), which will cover 99.75% of distinct mers and 56.63% of all mers.
Average number of kmers between count 1193 and 1244 is 6141803
Found break in kmer coverage at 139724
Set maxCount to 73422 (2 kmers), which will cover 100.00% of distinct mers and 90.00% of all mers.
```
I am not sure it has changed the threshold, is it maxCount value that I need to look at?

Also, the number of short reads < 5kb is very high as compared to the default canu run, is that expected because of the corOut and MerThreshold cutoffs?

**Comment by brianwalenz** [comment]
The two are the same; case doesn't matter.

Our logging here is, well, terrible; the ovlMerThreshold change isn't reported.  Check the sequence names in 0-mercounts/*frequentMers.fasta - if those numbers are smaller than what you set (the file should also be large), it worked.  If they're still enormous (and a small file) it didn't work.

You'll only see effects of this change when computing trimming and unitigging overlaps.

Finally, this is definitely the problem.  73,422 is a ridiculous number.

**Comment by ersgupta** [comment]
Ok.
I think it has not changed the threshold, could you please confirm? I will try with ```ovlMerThreshold``` (instead of ```OvlMerThreshold```) for the subsequent steps. 

```
$ grep "^>" canu.het.ms22.frequentMers.fasta |sed 's/^>//' > headers
$ wc -l headers
16378277
$ awk -F"\t" '{if($1 > 300) print }' headers | wc -l
16322690
$ awk -F"\t" '{if($1 > 500) print }' headers | wc -l
9978553
```

Also, the trimming step is complete now and I get 56.75x coverage. So is it ok, even if it did not change the threshold for the previous steps?

```
-- In gatekeeper store './canu.het.gkpStore':
--   Found 8001928 reads.
--   Found 79463774282 bases (56.75 times coverage).
--
--   Read length histogram (one '*' equals 9121.57 reads):
--        0    999      0 
--     1000   1999 638510 **********************************************************************
--     2000   2999 519235 ********************************************************
--     3000   3999 495724 ******************************************************
--     4000   4999 486333 *****************************************************
--     5000   5999 474913 ****************************************************
--     6000   6999 464859 **************************************************
--     7000   7999 453484 *************************************************
--     8000   8999 446387 ************************************************
--     9000   9999 435873 ***********************************************
--    10000  10999 437647 ***********************************************
--    11000  11999 436253 ***********************************************
--    12000  12999 400429 *******************************************
--    13000  13999 360294 ***************************************
--    14000  14999 316234 **********************************
--    15000  15999 277610 ******************************
--    16000  16999 240014 **************************
--    17000  17999 204094 **********************
--    18000  18999 171502 ******************
--    19000  19999 143161 ***************
--    20000  20999 118374 ************
--    21000  21999  97436 **********
--    22000  22999  79214 ********
--    23000  23999  64132 *******
--    24000  24999  52113 *****
--    25000  25999  41445 ****
--    26000  26999  33205 ***
--    27000  27999  26134 **
--    28000  28999  20429 **
--    29000  29999  16164 *
--    30000  30999  12491 *
--    31000  31999   9691 *
--    32000  32999   7417 
--    33000  33999   5576 
--    34000  34999   4099 
--    35000  35999   3248 
--    36000  36999   2337 
--    36000  36999   2337                                                                                                                                                                      --    37000  37999   1732 
--    38000  38999   1175 
--    39000  39999    936 
--    40000  40999    626 
--    41000  41999    453 
--    42000  42999    336 
--    43000  43999    196 
--    44000  44999    155 
--    45000  45999     85 
--    46000  46999     64 
--    47000  47999     34 
--    48000  48999     32 
--    49000  49999     15 
--    50000  50999     13 
--    51000  51999      4 
--    52000  52999      1 
--    53000  53999      2 
--    54000  54999      3 
--    55000  55999      1 
--    56000  56999      3
--    57000  57999      0 
--    58000  58999      0 
--    59000  59999      0 
--    60000  60999      0 
--    61000  61999      0 
--    62000  62999      0 
--    63000  63999      1 
```

**Comment by brianwalenz** [comment]
I probably got the logic wrong in my original explanation.  A setting of ovlMerThreshold=300 will result in a fasta file with kmers with count greater than 300.  Your check for both 300 and 500 was perfect!  It showed you have kmers with count between 300 and 500, exactly what we wanted to see.

This is really just a speed optimization.  It tells the overlapper to not look for overlaps using seeds (kmers) that are excessively common in the data.  Trimming won't change much.  Assembly _could_ change, in that semi-common repeats might not assemble.  We haven't explored the impact of this much, but expect it'd take a very low threshold to make an important difference.

**Comment by ersgupta** [comment]
Got it. My main concern has just been the runtime for unitigging overlap (> 1 month). I will let you know what threshold it sets this time for unitigging. I am using ovlMerThreshold 500 for unitigging to assemble "some" semi-common repeats.

**Comment by ersgupta** [comment]
Command used: 
It seems it is not working, could you please confirm?

```
$ cat unitigging/0-mercounts/canu.het.ms22.estMerThresh.err 
RAW MER COUNTS:
  distinct:   5880697710 (different kmer sequences)
  unique:     4139013039 (single-copy kmers)
  total:     79288429521 (kmers in sequences)

Set maxCount to 1223 (4932 kmers), which will cover 99.75% of distinct mers and 56.71% of all mers.
Average number of kmers between count 1198 and 1249 is 6034210
Found break in kmer coverage at 150377
Set maxCount to 72836 (3 kmers), which will cover 100.00% of distinct mers and 90.00% of all mers.
$ cat unitigging/0-mercounts/canu.het.ms22.estMerThresh.out 
72836
```

Can I manually change the value in ```unitigging/0-mercounts/canu.het.asm.ms22.estMerThresh.out``` to 500 and run canu with ```-assemble``` with the trimmed reads?

**Comment by brianwalenz** [comment]
How long are jobs taking now?

**Comment by ersgupta** [comment]
Extremely fast. All got done in 2 days. :) I was waiting for unitigging to get over to reply here. Right now Here are the stats till now:
```
-- Found 743 overlapInCore output files.
--
-- overlapInCore compute 'unitigging/1-overlapper':
--   kmer hits
--     with no overlap     128625358382  229.316285 +- 62326426.777
--     with an overlap       5311854373  8.34858681 +- 2621489.327
--
--   overlaps                5311854373  8.34858681 +- 2621489.327
--     contained             2117448745  3.72139973 +- 1062236.745
--     dovetail              3194405628  4.62718708 +- 1563520.792
--
--   overlaps rejected
--     multiple per pair              0           0 +- 0
--     bad short window               0           0 +- 0
--     bad long window                0           0 +- 0
----------------------------------------
...
-- Overlap store 'unitigging/canu.het.asm.ovlStore' contains:
--
--   category            reads     %          read length        feature size or coverage  analysis
--   ----------------  -------  -------  ----------------------  ------------------------  --------------------
--   middle-missing         94    0.00    13064.30 +- 8970.81       1133.16 +- 2440.94    (bad trimming)
--   middle-hump             2    0.00     3673.00 +- 695.79        1887.50 +- 992.07     (bad trimming)
--   no-5-prime             41    0.00     5979.78 +- 6533.08        386.59 +- 616.07     (bad trimming)
--   no-3-prime             38    0.00     6808.16 +- 7291.49        313.87 +- 586.65     (bad trimming)
--   
--   low-coverage         2792    0.03     3743.94 +- 3166.44         11.12 +- 4.28       (easy to assemble, potential for lower quality consensus)
--   unique            2778010   34.72     8644.79 +- 5716.54         54.50 +- 16.41      (easy to assemble, perfect, yay)
--   repeat-cont       1686391   21.07     6181.41 +- 5220.53       3022.57 +- 2650.13    (potential for consensus errors, no impact on assembly)
--   repeat-dove          3585    0.04    24696.91 +- 8660.68       1338.88 +- 1527.14    (hard to assemble, likely won't assemble correctly or even at all)
--   
--   span-repeat        501255    6.26    14640.04 +- 6805.16       4201.00 +- 4227.77    (read spans a large repeat, usually easy to assemble)
--   uniq-repeat-cont  2085037   26.06    11114.09 +- 5084.96                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)
--   uniq-repeat-dove   176069    2.20    23103.87 +- 5436.75                             (will end contigs, potential to misassemble)
--   uniq-anchor        768613    9.61    13457.89 +- 5858.75       6749.11 +- 5461.57    (repeat read, with unique section, probable bad read)
--
-- Loading read lengths.
-- Loading number of overlaps per read.
```

Right now it is at oea.

**Comment by brianwalenz** [comment]
Yay!  Reasonable number of jobs there?

**Comment by ersgupta** [comment]
You mean the number of overlap jobs? I think the number of jobs is more this time, probably because of coverage? But the speed gain is substantial with ovlMerThreshold.

The stats to assemble are good, with ovlMerThreshold cutoff, right?

**Comment by brianwalenz** [comment]
Nope, I was interested in the number of RED and/or OEA jobs - the step you're in now.  I'm assuming that no complaints mean it's all good, but just checking before I closed the issue.

The stats look good.  Lots of repeats in there, but we kind of knew that already.  The repeats might lead to false contig splitting, but no way to know until it's all done.

**Comment by ersgupta** [comment]
Ah ok. RED is done: 1197 jobs. OEA is running with 266 jobs configured with 8gb per job. I will post the stats once the assembly is complete. I expect another day or 2.

The good news is, with ovlMerThreshold (atleast in my case), I can finish one canu run in 2-3 weeks, which is a quite reasonable time.

**Comment by brianwalenz** [comment]
Next release should have a fix for the threshold.

I'll close, but feel free to update with stats, or open a new one if it's terrible.

**Comment by ersgupta** [comment]
Sure. Thanks 👍 :)

**Comment by ersgupta** [comment]
So the assembly is over with a huge improvement in assembly from the default run (current assembly is with het parameters). Next I need to separate the primary contigs, as also indicated by BUSCO duplicate %age.
BUSCO:
```
embryophyta:
C:90.1%[S:39.0%,D:51.1%],F:2.1%,M:7.8%,n:1440
eukaryota:
C:88.1%[S:28.7%,D:59.4%],F:1.0%,M:10.9%,n:303
```

```
[UNITIGGING/CONTIGS]
-- Found, in version 1, after unitig construction:
--   contigs:      13136 sequences, total length 1945343614 bp (including 1199 repeats of total length 48146570 bp).
--   bubbles:      0 sequences, total length 0 bp.
--   unassembled:  945759 sequences, total length 6961558048 bp.
--
-- Contig sizes based on genome size --
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10     3113192            34   141811479
--     20     2158729            89   280413198
--     30     1677870           164   420522108
--     40     1282178           259   560192504
--     50      971966           385   700091206
--     60      726293           551   840011822
--     70      518958           780   980377878
--     80      354122          1108  1120340329
--     90      237539          1596  1260091985
--    100      156910          2330  1400117645
--    110      102175          3446  1540094619
--    120       60897          5238  1680025986
--    130       36676          8254  1820005030
--

[UNITIGGING/CONSENSUS]
-- Found, in version 2, after consensus generation:
--   contigs:      13136 sequences, total length 1942743071 bp (including 1199 repeats of total length 48079663 bp).
--   bubbles:      0 sequences, total length 0 bp.
--   unassembled:  945759 sequences, total length 6961550311 bp.
--
-- Contig sizes based on genome size --
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10     3109544            34   141500347
--     20     2148278            90   281963413
--     30     1671973           165   421296386
--     40     1278079           260   560299768
--     50      966921           387   700569720
--     60      720846           554   840509047
--     70      515621           783   980029102
--     80      352491          1113  1120032692
--     90      235518          1605  1260027568
--    100      155125          2345  1400119365
--    110      101021          3469  1540006991
--    120       60466          5277  1680008568
--    130       36380          8317  1820006634
```

Default run:
```
[UNITIGGING/CONTIGS]
-- Found, in version 1, after unitig construction:
--   contigs:      18521 sequences, total length 1961109191 bp (including 3695 repeats of total length 115653645 bp).
--   bubbles:      0 sequences, total length 0 bp.
--   unassembled:  274950 sequences, total length 2853476366 bp.
--
-- Contig sizes based on genome size --
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10     1263479            78   140154522
--     20      869261           213   280005764
--     30      660536           400   420390458
--     40      514012           642   560406863
--     50      396314           954   700044636
--     60      300769          1362   840017394
--     70      229940          1896   980054255
--     80      177514          2593  1120083243
--     90      136170          3494  1260049556
--    100      101754          4686  1400045773
--    110       71917          6321  1540008569
--    120       49278          8695  1680026340
--    130       33385         12160  1820010175
--    140        7564         18278  1960002950
--

[UNITIGGING/CONSENSUS]
-- Found, in version 2, after consensus generation:
--   contigs:      18521 sequences, total length 1962338174 bp (including 3695 repeats of total length 115813290 bp).
--   bubbles:      0 sequences, total length 0 bp.
--   unassembled:  274950 sequences, total length 2853470289 bp.
--
-- Contig sizes based on genome size --
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10     1262025            78   140154584
--     20      870162           213   280096733
--     30      660588           400   420541211
--     40      514990           641   560118368
--     50      396734           954   700342850
--     60      301164          1361   840106537
--     70      230076          1894   980090307
--     80      178007          2590  1120140665
--     90      136302          3489  1260067142
--    100      102061          4678  1400082861
--    110       72368          6307  1540022998
--    120       49445          8671  1680019474
--    130       33534         12121  1820009248
--    140        9546         18134  1960008149
```
